### DISCUSSION TITLE
Milvus cosine similarity not working with pymilvus==2.3.1

### DISCUSSION THREAD

**Discussion by rajendra-t** [milvus-io/milvus#28203] [Q&A and General discussion]
hi All, I am using pymilvus 2.3.1 and below code to Index the data.
index_params = {
    'metric_type': 'COSINE',
    'index_type': "IVF_FLAT",
    'params': {"nlist": 1024}
}

However I am getting below error
RPC error: [create_index], <MilvusException: (code=1, message=metric type not found or not supported, supported: [L2 IP])>, <Time:{'RPC start': '2023-11-06 16:04:56.994282', 'RPC error': '2023-11-06 16:04:57.316788'}>

We've deployed v2.3.2 of Milvus operator on k8 cluster

Can anyone please suggest how to make cosine similarity work

**Comment by xiaofan-luan** [comment]
IVF doesn't support Cosine for now. 2.3.2 should solve the problem? @liliu-z

**Comment by yhmo** [comment]
@rajendra-t 
I guess your milvus version is not v2.3.2.
I just use a milvus v2.3.2 to test IVF_FLAT with COSINE metric type, it works well.
You can use get_server_version() to get the server version.
`print(utility.get_server_version())`

**Comment by rajendra-t** [comment]
Thanks @yhmo for the inputs, yes the Milvus server version is the issue - will try to validate this with 2.3.2 version of Milvus server
